test/libsolidity/util/TestFileParser.cpp

```diff
@@ -589,8 +615,18 @@ void TestFileParser::Scanner::scanNextToken()
 				BOOST_THROW_EXCEPTION(TestParserError("Unexpected character: '" + string{current()} + "'"));
 			break;
 		}
+	} while (m_currentToken == Token::Whitespace);
+}
+
+string TestFileParser::Scanner::scanEffectString()
+{
+	string effect;
+	while (peek() != '/' && peek() != '\0')
```

Comment:
What is the meaning of `/`? Shouldn't this scan for newline or something?

Reply:
Basically the meaning of `/` is newline - I know that this not perfectly nice. It looks like that newlines got removed.

Reply:
What about effects that contain `/` character somewhere in the middle?

Reply:
I think the best is to just disallow`/` in side-effect strings.

Reply:
After #11391, I think this here should be just `while (!eof() && peek() != '\n')` for readability.

Reply:
Could also just add a `string Scanner::readLine()` helper which does this, perhaps makes it more clear? Or just rename this to `readLine()` in fact may be better.